lib/clusterManager.js

```diff
@@ -189,7 +194,64 @@ if (cluster.isMaster) {
     this.overrideFileConfig = override;
   }
 
-   ClusterManager.prototype.notifyWorkersForAddingPlugin = function(resultHandler, indexInCluster) {
+  /* Having gotten here, we are already at the cluster, but it's safer to validate anyway
+    i. e. "if (this.storage) ..." since only the master cluster has storage */
+  ClusterManager.prototype.getStorage = function(resultHandler) {
+    if (this.storage) {
+      return resultHandler(this.storage);
+    } else {
+      return resultHandler(zluxUtil.DataserviceStorage());
+    }
+  }
+
+  ClusterManager.prototype.setStorageValue = function(id, value, resultHandler) {
+    if (this.storage) {
+      this.storage.setStorageValue(id, value);
+      return resultHandler(true);
+    } else {
+      clusterLogger.warn("ZWED0160W - Master cluster not found to save storage for id: " + id + ", value: " + value);
```

Comment:
The cluster will have storage because it is made upon startup. The plugins will always have storage because even before cluster storage is checked for, a default local one is made, so no plugin functionality should break. If we create the storage after encountering an error, then it may be because this method was called at startup, so setStorage for an object that doesn't exist yet will turn up false and creating a storage object here would be creating a second object and thus mess with things terribly